### PR TITLE
fix: remove moment patches

### LINKED ISSUE #27963: TypeError: Cannot read properties of undefined (reading 'parsedDateParts')
Sentry Issue: [APPSMITH-74S](https://appsmith.sentry.io/issues/4508152285/?referrer=github_integration)

```
TypeError: Cannot read properties of undefined (reading 'parsedDateParts')
  at Kr (reducers/evaluationReducers/treeReducer.ts:46:11)
  at reducer (utils/ReducerUtils.ts:23:14)
  at next (RouteParamsMiddleware.ts:125:12)
  at <object>.value (utils/WorkerUtil.ts:213:12)
...
(42 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by vsvamsi1**
## Description
Updates were trying to be added it to invalid moment objects which is not necessary. Earlier when we serialse invalid Moment objects it gets converted to null, made code corrections which it to keep it inline with earlier behaviour.

#### PR fixes following issue(s)
Fixes #27963

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [x] Manual
- [ ] JUnit
- [x] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [x] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/ListV2_SerververSide_spec.js [hunk 1]
```diff
@@ -46,11 +46,16 @@ describe("List widget V2 Serverside Pagination", () => {
   it("2. Next button disabled but visible in view mode when there's no data", () => {
     agHelper.AssertText(commonlocators.listPaginateActivePage, "text", "1");
     agHelper.GetNClick(commonlocators.listPaginateNextButton, 0, true);
+    cy.wait(1000);
```
Comment (ohansFavour) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/ListV2_SerververSide_spec.js:49]: Could you help me understand why we are adding delay here?

File: app/client/src/workers/Evaluation/helpers.ts [hunk 2]
```diff
@@ -201,6 +202,20 @@ const generateDiffUpdates = (
 
       const lhs = get(oldDataTree, segmentedPath);
 
+      //convert all invalid moment objects to nulls ...
+      //large collect nodes are anyway getting serialised so the invalid objects will be converted to nulls
+      if (isMoment(rhs) && !rhs.isValid()) {
```
Comment (ohansFavour) [inline app/client/src/workers/Evaluation/helpers.ts:207]: Is this issue peculiar to only moment object RHS?.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/ListV2_SerververSide_spec.js (hunk 1 above)
Comment (vsvamsi1) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/ListV2_SerververSide_spec.js:49]: This test case is failing consistently in CI but passing locally, after this delay it starts passing in CI. Im not sure why.

File: app/client/src/workers/Evaluation/helpers.ts (hunk 2 above)
Comment (vsvamsi1) [reply app/client/src/workers/Evaluation/helpers.ts:207]: I have detected this issue for invalid moment objects through CI. In sentry most of the errors are related to invalid moment object, im not sure if there are other cases as well. We can see post merging if it brings down the number of sentry issues.

**Review by ohansFavour — APPROVED** [review]

**Comment by ramsaptami** [comment]
lgtm
cc: @vsvamsi1

**Pull request merged**